src/iter/collect/mod.rs

```diff
@@ -90,7 +90,11 @@ impl<'c, T: Send + 'c> Collect<'c, T> {
         F: FnOnce(CollectConsumer<'_, T>) -> CollectResult<'_, T>,
     {
         let slice = Self::reserve_get_tail_slice(&mut self.vec, self.len);
-        let result = scope_fn(CollectConsumer::new(slice));
+        let result = scope_fn(CollectConsumer::new(
+            slice.as_mut_ptr().cast(),
+            slice.len(),
+            std::marker::PhantomData,
```

Comment:
This marker is meant to tie the lifetime, but here we're producing it out of thin air, so it could be anything.